Doc 12410 319 maint release

ElliotFrancisHunter opened the pull request:
Documentation for 3.1.9 maintenance release.

Includes:

- Database Fullsync API update docs (In database common)
- Snippets required for above feature
- Standard release items (Update antora.yaml Release notes, update to gs-downloads for C)

ElliotFrancisHunter committed:
DOC-12410: standard maint release items

Updated antora to new version for all platforms
Added release notes for all platforms
Added new version to gs-downloads for c

ElliotFrancisHunter committed:
DOC-12410: Added full sync common copy

Added docs copy to common file shared across platforms.

ElliotFrancisHunter committed:
DOC-12410: Fullsync snippets

Snippets for all platforms

borrrden commented on modules/csharp/examples/code_snippets/Program.cs:
```diff
@@ -449,6 +449,15 @@ private static void CloseDatabase()
+        private static void DatabaseFullsync()
+        {
+           // tag::database-fullsync[] 
+          var config = new DatabaseConfiguration();
```
This part is probably redundant.  You can save some space by not including it in the tag.

ElliotFrancisHunter replied:
Removed.

borrrden commented on modules/csharp/examples/code_snippets/Program.cs:
```diff
@@ -449,6 +449,15 @@ private static void CloseDatabase()
+           // tag::database-fullsync[] 
+          var config = new DatabaseConfiguration();
+          // this enables fullsync
+          config.FullSync = false;
```
Misleading comment.  `false` is not going to enable anything.

ElliotFrancisHunter replied:
Good spot, changed to true.

borrrden commented on modules/objc/examples/code_snippets/SampleCodeTest.m:
```diff
@@ -205,6 +213,9 @@ - (void) dontTestDatabaseEncryption {
     // tag::database-encryption[]
     CBLDatabaseConfiguration *config = [[CBLDatabaseConfiguration alloc] init];
     config.encryptionKey = [[CBLEncryptionKey alloc] initWithPassword:@"secretpassword"];
+    // tag::database-fullsync[]
```
You added this tag twice...

ElliotFrancisHunter replied:
Good spot! Was fiddling with tag behaviour and it didn't work out. Removed dead code.

borrrden committed:
Update C# validation to pull specific version

ElliotFrancisHunter committed:
DOC-12410: Snippet fixes

Removed redundant line from csharp snippet and changed value of FullSync to true to match comment content.
Dead code removal in objc snippet to remove duplicate content

RayOffiah commented on modules/ROOT/pages/_partials/commons/common-database.adoc:
```diff
@@ -136,6 +136,21 @@ endif::is-c[]
+This feature is not enabled by default and must be manually set in your database configuration.
+
+
+CAUTION: Database Full Sync is a safe method to preserve data loss but will incur a significant degredation of performance.
```
There's a CAUTION admonition??
You learn something new every day!

velicuvlad commented on modules/csharp/examples/code_snippets/Program.cs:
```diff
@@ -452,9 +452,8 @@ private static void CloseDatabase()
         private static void DatabaseFullsync()
         {
            // tag::database-fullsync[] 
-          var config = new DatabaseConfiguration();
```
this can't be completely removed, otherwise config is invalid when trying to set `.FullSync`. As Jim pointed and from slack, having just `config.FullSync = true` will suffice.

velicuvlad commented on modules/android/examples/codesnippet_collection.java:
```diff
@@ -632,14 +632,21 @@ public void newDatabaseExample() throws CouchbaseLiteException {
 
+     public void databaseFullSyncExample() throws CouchbaseLiteException {
+        // tag::database-fullsync[]
+        DatabaseConfiguration config = new DatabaseConfiguration();
```
similar, move tags

ElliotFrancisHunter replied:
Done!

velicuvlad commented on modules/java/examples/codesnippet_collection.java:
```diff
@@ -617,6 +617,13 @@ public void newDatabaseExample() throws CouchbaseLiteException {
 
+     public void DatabaseFullSyncExample() throws CouchbaseLiteException {
+        // tag::database-fullsync[]
+        DatabaseConfiguration config = new DatabaseConfiguration();
```
similar, move tags

ElliotFrancisHunter replied:
Done

RayOffiah reviewed (approved):
I'll remember [CAUTION] :-)

velicuvlad commented on modules/csharp/examples/code_snippets/Program.cs:
```diff
@@ -449,6 +449,14 @@ private static void CloseDatabase()
+        private static void DatabaseFullsync()
+        {
+           // tag::database-fullsync[] 
+           // this enables fullsync
```
Test build is failing because `config` is not set. You have to
`var config = new DatabaseConfiguration()`, just put it before the `// tag:`.

ElliotFrancisHunter replied:
Noted, done!

velicuvlad commented on modules/objc/examples/code_snippets/SampleCodeTest.m:
```diff
@@ -187,6 +187,14 @@ - (void) dontTestNewDatabase {
 
+- (void) dontTestDatabaseFullSync {
+  // tag::database-fullsync[]
+  CBLDatabaseConfiguration* config = [[CBLDatabaseConfiguration alloc] init];
```
same here

ElliotFrancisHunter replied:
Done

velicuvlad commented on modules/swift/examples/code_snippets/SampleCodeTest.swift:
```diff
@@ -60,6 +60,16 @@ class SampleCodeTest {
 
+    func dontTestDatabaseFullSync() throws {
+        // tag::database-fullsync[]
+        var config = DatabaseConfiguration()
```
here as well

ElliotFrancisHunter replied:
Done

velicuvlad commented on modules/android/examples/codesnippet_collection.kt:
```diff
@@ -120,6 +120,22 @@ class BasicExamples(private val context: Context) {
         database.delete()
     }
 
+    // ### Database FullSync
+    fun DatabaseFullSyncExample() {
+        // tag::database-fullsync[]
+        val database = Database(
+            "my-db",
+            DatabaseConfigurationFactory.newConfig(
+                fullSync = true
+            )
+        ) 
+        // end::database-fullsync[]
+        
+        config.setFullSync()
+        
+    }
```
I am unsure of how correct this is. I would wait on Blake to review Android/Kotlin

pasin replied:
> config.setFullSync()

Seem like it can be just deleted.

bmeike replied:
This is correct.  It is also cute.  Can we please keep it?

ElliotFrancisHunter committed:
Doc-12410: Snippet fixes

Added initaliser for database config into .NET snippet
Moved tags to remove unnecessary info

velicuvlad commented on Jenkinsfile:
please update C to 3.1.9, same as Jim did for C#. This will fix the Jenkins.

ElliotFrancisHunter replied:
Done

ElliotFrancisHunter committed:
DOC-12410: Tidy snippet tag

ElliotFrancisHunter committed:
DOC-12410: Update c build to correct version

This should fix failing build.

ElliotFrancisHunter committed:
DOC-12410: Dead code removal

ElliotFrancisHunter committed:
DOC-12410: Temporary workaround for GA

Kotlin snippet tag removed until after GA so we can publish correct snippet post review from dev team

ElliotFrancisHunter commented:
cc: @idulo tags have been removed so Kotlin snippet is not rendered. This can be rectified post GA if you're planning on pushing this out before I return from PTO on Wednesday.
